components/apc_ups/apc_ups.h

```diff
@@ -114,15 +119,34 @@ class ApcUps : public uart::UARTDevice, public PollingComponent {
   APC_UPS_BINARY_SENSOR(output_overloaded, Q, Q)
   APC_UPS_BINARY_SENSOR(battery_low, Q, Q)
   APC_UPS_BINARY_SENSOR(replace_battery, Q, Q)
+  APC_UPS_SENSOR(ambient_humidity, LOWER_H, h, float)
+  APC_UPS_SENSOR(return_threshold, LOWER_E, e, float)
   APC_UPS_SENSOR(state_of_charge, LOWER_F, f, float)
+  APC_UPS_SENSOR(nominal_battery_voltage, LOWER_G, g, float)
+  APC_UPS_SENSOR(estimated_runtime, LOWER_J, j, float)
+  APC_UPS_SENSOR(low_transfer_voltage, LOWER_L, l, float)
+  APC_UPS_SENSOR(nominal_output_voltage, LOWER_O, o, float)
+  APC_UPS_SENSOR(upper_transfer_voltage, LOWER_U, u, float)
 
+  APC_UPS_VALUED_TEXT_SENSOR(old_firmware_version, V, V, std::string)
+  APC_UPS_VALUED_TEXT_SENSOR(self_test_results, X, X, std::string)
   APC_UPS_VALUED_TEXT_SENSOR(protocol_info, LOWER_A, a, std::string)
   APC_UPS_VALUED_TEXT_SENSOR(firmware_revision, LOWER_B, b, std::string)
   APC_UPS_VALUED_TEXT_SENSOR(local_identifier, LOWER_C, c, std::string)
-  APC_UPS_SENSOR(estimated_runtime, LOWER_J, j, float)
+  APC_UPS_VALUED_TEXT_SENSOR(alarm_delay, LOWER_K, k, std::string)
   APC_UPS_VALUED_TEXT_SENSOR(manufacture_date, LOWER_M, m, std::string)
+  APC_UPS_VALUED_TEXT_SENSOR(serial_number, LOWER_N, n, std::string)
   APC_UPS_SENSOR(ambient_temperature, LOWER_T, t, float)
+  APC_UPS_VALUED_TEXT_SENSOR(measure_upc_firmware, LOWER_V, v, std::string)
   APC_UPS_VALUED_TEXT_SENSOR(last_battery_change_date, LOWER_X, x, std::string)
+  APC_UPS_VALUED_TEXT_SENSOR(copyright_notice, LOWER_Y, Y, std::string)
+  APC_UPS_VALUED_TEXT_SENSOR(line_quality, 9, 9, std::string)
+  APC_UPS_VALUED_TEXT_SENSOR(model_name, CTRL_A, "\x01", std::string)
```

Comment:
@samoswall Did you see this simplification? The special handling isn't required anymore.

Reply:
I didn't notice the simplification. I'm not a very strong programmer :)